app/main.py

```diff
@@ -86,128 +86,51 @@
 }
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def apply_effects(knight: dict) -> None:
+    """Apply all effects (armor, weapon, potion) to a knight."""
+    # Initialize protection
+    knight["protection"] = 0
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    # Apply armor
+    for armor_piece in knight["armour"]:
+        knight["protection"] += armor_piece["protection"]
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+    # Apply weapon
+    knight["power"] += knight["weapon"]["power"]
 
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
+    # Apply potion if exists
+    if knight["potion"] is not None:
+        for stat, value in knight["potion"]["effect"].items():
+            knight[stat] += value
```

Comment:
The code assumes that all potion effects directly modify the knight's attributes. However, the 'protection' effect should be added to 'knight["protection"]', not directly to 'knight["protection"]'. Consider checking if the stat is 'protection' and handling it separately.